bug: Funding schedule is one day early.

![image](https://user-images.githubusercontent.com/37967690/192118823-8410b20d-36ff-4338-9c2c-1845d1f59e0a.png)

```go
t.Run("regression", func(t *testing.T) {
	rule := testutils.Must(t, models.NewRule, "FREQ=MONTHLY;INTERVAL=1;BYMONTHDAY=15,-1")
	timezone := testutils.Must(t, time.LoadLocation, "America/Chicago")
	next := time.Date(2022, 9, 15, 0, 0, 0, 0, timezone)
	expected := time.Date(2022, 9, 30, 0, 0, 0, 0, timezone)
	// 1 Second after midnight in timezone on last funding day. But in UTC because that's the server timezone.
	now := time.Date(2022, 9, 15, 0, 0, 1, 0, timezone).UTC()
	fundingSchedule := models.FundingSchedule{
		Name:            "Payday",
		Rule:            rule,
		ExcludeWeekends: false,
		LastOccurrence:  nil,
		NextOccurrence:  next,
	}
	nextFundingOccurrence := fundingSchedule.GetNextContributionDateAfter(now, timezone)
	assert.Equal(t, expected, nextFundingOccurrence, "should be on friday the 30th of september next")
})
```
^ Test above proves the bug.

```
=== RUN   TestFundingSchedule_CalculateNextOccurrence
--- FAIL: TestFundingSchedule_CalculateNextOccurrence (0.00s)
=== RUN   TestFundingSchedule_CalculateNextOccurrence/regression
    funding_schedule_test.go:92: 
        	Error Trace:	/Users/elliotcourant/monetr/monetr/pkg/models/funding_schedule_test.go:92
        	Error:      	Not equal: 
        	            	expected: time.Date(2022, time.September, 30, 0, 0, 0, 0, time.Location("America/Chicago"))
        	            	actual  : time.Date(2022, time.September, 29, 0, 0, 0, 0, time.Location("America/Chicago"))
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,2 +1,2 @@
        	            	-(time.Time) 2022-09-30 00:00:00 -0500 CDT
        	            	+(time.Time) 2022-09-29 00:00:00 -0500 CDT
        	            	 
        	Test:       	TestFundingSchedule_CalculateNextOccurrence/regression
        	Messages:   	should be on friday the 30th of september next
    --- FAIL: TestFundingSchedule_CalculateNextOccurrence/regression (0.00s)
```

--- a/pkg/models/funding_schedule_test.go
+++ b/pkg/models/funding_schedule_test.go
@@ -131,4 +131,22 @@ func TestFundingSchedule_GetNextContributionDateAfter(t *testing.T) {
 		next := fundingSchedule.GetNextContributionDateAfter(now, time.UTC)
 		assert.Equal(t, expected, next, "should not show the 15th, instead should show the 31st")
 	})
+
+	t.Run("prevent regression calculating midnight", func(t *testing.T) {
+		rule := testutils.Must(t, models.NewRule, "FREQ=MONTHLY;INTERVAL=1;BYMONTHDAY=15,-1")
+		timezone := testutils.Must(t, time.LoadLocation, "America/Chicago")
+		next := time.Date(2022, 9, 15, 0, 0, 0, 0, timezone)
+		expected := time.Date(2022, 9, 30, 0, 0, 0, 0, timezone)
+		// 1 Second after midnight in timezone on last funding day. But in UTC because that's the server timezone.
+		now := time.Date(2022, 9, 15, 0, 0, 1, 0, timezone).UTC()
+		fundingSchedule := models.FundingSchedule{
+			Name:            "Payday",
+			Rule:            rule,
+			ExcludeWeekends: false,
+			LastOccurrence:  nil,
+			NextOccurrence:  next,
+		}
+		nextFundingOccurrence := fundingSchedule.GetNextContributionDateAfter(now, timezone)
+		assert.Equal(t, expected, nextFundingOccurrence, "should be on friday the 30th of september next")
+	})
 }

--- a/pkg/util/midnight.go
+++ b/pkg/util/midnight.go
@@ -21,7 +21,7 @@ func MidnightInLocal(input time.Time, timezone *time.Location) time.Time {
 		input.Minute(),
 		input.Second(),
 		input.Nanosecond(),
-		timezone,
+		time.UTC,
 	)
 	// We need to do this because we need to know the offset for a given timezone at the provided input's timestamp. This
 	// way we can adjust the input to be in that timezone then truncate the time.

--- a/pkg/util/midnight_test.go
+++ b/pkg/util/midnight_test.go
@@ -5,6 +5,7 @@ import (
 	"time"
 
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 )
 
 func TestInLocal(t *testing.T) {
@@ -29,22 +30,13 @@ func TestInLocal(t *testing.T) {
 }
 
 func TestMidnightInLocal(t *testing.T) {
-	t.Run("late in the day", func(t *testing.T) {
-		central, err := time.LoadLocation("America/Chicago")
-		assert.NoError(t, err, "must be able to load the central time location")
-		start := time.Date(2022, 06, 15, 2, 24, 38, 0, time.UTC)
-
-		midnight := MidnightInLocal(start, central)
-		assert.Equal(t, time.Date(2022, 06, 14, 0, 0, 0, 0, central), midnight, "midnight in a different timezone is a different day")
-	})
-
 	t.Run("weird timezone", func(t *testing.T) {
 		sanLuis, err := time.LoadLocation("America/Argentina/San_Luis")
 		assert.NoError(t, err, "must be able to load the sanLuis time location")
 		start := time.Date(2022, 06, 15, 2, 24, 38, 0, time.UTC)
 
 		midnight := MidnightInLocal(start, sanLuis)
-		assert.Equal(t, time.Date(2022, 06, 14, 0, 0, 0, 0, sanLuis), midnight, "midnight in a different timezone is a different day")
+		assert.Equal(t, time.Date(2022, 06, 15, 0, 0, 0, 0, sanLuis), midnight, "midnight in a different timezone is a different day")
 	})
 
 	t.Run("panics for an empty time", func(t *testing.T) {
@@ -54,5 +46,16 @@ func TestMidnightInLocal(t *testing.T) {
 		assert.Panics(t, func() {
 			_ = MidnightInLocal(time.Time{}, sanLuis)
 		})
+	})
+
+	t.Run("central time", func(t *testing.T) {
+		timezone, err := time.LoadLocation("America/Chicago")
+		require.NoError(t, err, "must load central timezone")
+		// 12 seconds after midnight already in the desired timezone.
+		input := time.Date(2022, 9, 15, 0, 0, 12, 0, timezone)
+		expected := time.Date(2022, 9, 15, 0, 0, 0, 0, timezone)
+
+		midnight := MidnightInLocal(input, timezone)
+		assert.Equal(t, expected, midnight, "should have truncated the time, but not the timezone")
 	})
 }

